packages/forms/signals/src/api/metadata.ts

```diff
@@ -143,6 +148,16 @@ export function andMetadataKey(): AggregateMetadataKey<boolean, boolean> {
   );
 }
 
+export function resourceMetadataKey<TParams, TResult>(
+  factory: (params: Signal<TParams | undefined>) => ResourceRef<TResult>,
```

Comment:
We probably want a plain `Resource` here:

```suggestion
  factory: (params: Signal<TParams | undefined>) => Resource<TResult>,
```

`ResourceRef` gives powers to e.g. destroy the resource, which we don't want to expose to consumers.

Reply:
I removed this because it wasn't actually used anywhere (remnant from previous iteration of the change). I could add it back if we want it, but right now its just a matter of call `createWrappedMetadataKey((...) => resource(...))`